compliance/collection/nodeinventorizer/caching_inventorizer_test.go

```diff
@@ -0,0 +1,128 @@
+	return &storage.NodeInventory{NodeName: nodeName}, nil
+}
+
+type TestComplianceCachingSuite struct {
+	suite.Suite
+}
+
+func TestComplianceCaching(t *testing.T) {
+	suite.Run(t, new(TestComplianceCachingSuite))
+}
+
+// run before each test
+// func (s *TestComplianceCachingSuite) SetupTest() {
+//	s.sleeper = mockSleeper{callCount: 0}
+//	s.cs = *NewCachingScanner(
+//		fmt.Sprintf("%s/inventory-cache", s.T().TempDir()),
+//		s.sleeper.mockWaitCallback,
+//	)
+//}
+
+func (s *TestComplianceCachingSuite) TestMin() {
+	initial := 2 * time.Second
+	maxBackoff := 10 * time.Second
+
+	actual := min(initial, maxBackoff)
+
+	// On successful test, the duration must not be overwritten
+	s.Equal(initial, actual)
+}
+
+func (s *TestComplianceCachingSuite) TestMinMaxBackoff() {
+	initial, _ := time.ParseDuration("100h")
+	maxBackoff, _ := time.ParseDuration("10s")
+
+	actual := min(initial, maxBackoff)
+
+	s.Equal(maxBackoff, actual)
+}
+
+func (s *TestComplianceCachingSuite) TestCalcNextBackoff() {
+	initial, _ := time.ParseDuration("2s")
+	cache := initial
+	maxBackoff, _ := time.ParseDuration("10s")
+	cs := *NewCachingScanner(mockScanner{}, "", cache, initial, maxBackoff, func(time.Duration) {})
+	expectedBackoff := initial * backoffMultiplier
+
+	newBackoff := cs.calcNextBackoff(initial)
+
+	s.Equal(expectedBackoff, newBackoff)
+}
+
+func (s *TestComplianceCachingSuite) TestCalcNextBackoffUpperBoundary() {
+	initial, _ := time.ParseDuration("8s")
+	cache := initial
+	maxBackoff, _ := time.ParseDuration("10s")
+	cs := *NewCachingScanner(mockScanner{}, "", cache, initial, maxBackoff, func(time.Duration) {})
+
+	newBackoff := cs.calcNextBackoff(initial)
+
+	s.Equal(maxBackoff, newBackoff)
+}
+
+func (s *TestComplianceCachingSuite) TestTriggerNodeInventoryWithoutResultCache() {
+	initial, _ := time.ParseDuration("8s")
+	cache := initial
+	maxBackoff, _ := time.ParseDuration("10s")
+	inventoryCachePath := fmt.Sprintf("%s/inventory-cache", s.T().TempDir())
+	cs := *NewCachingScanner(mockScanner{}, inventoryCachePath, cache, initial, maxBackoff, func(time.Duration) {})
+	nodeName := "testme"
+
+	actual, e := cs.Scan(nodeName)
+
+	s.NoError(e)
+	s.Equal(nodeName, actual.GetNodeName())
+}
+
+func (s *TestComplianceCachingSuite) TestTriggerNodeInventoryHonorBackoff() {
+	sleeper := mockSleeper{callCount: 0}
+	expectedDuration, _ := time.ParseDuration("4s")
+
+	initial, _ := time.ParseDuration("2s") // must be lower than BackoffDuration in file
+	cache := initial
+	maxBackoff, _ := time.ParseDuration("10s")
+	inventoryCachePath := fmt.Sprintf("%s/inventory-cache", s.T().TempDir())
+	cs := *NewCachingScanner(mockScanner{}, inventoryCachePath, cache, initial, maxBackoff, sleeper.mockWaitCallback)
+
+	w := inventoryWrap{
+		CacheValidUntil:      time.Time{},
+		RetryBackoffDuration: 4000000000, // 4 seconds
+		CachedInventory:      "",
+	}
+	jsonWrap, e := json.Marshal(&w)
+	s.NoError(e)
+	e = os.WriteFile(cs.inventoryCachePath, jsonWrap, 0600)
+	s.NoError(e)
```

Comment:
I suppose you may have to use this fragment frequently in the test so you may want to create a helper function in `TestComplianceCachingSuite` that hides burden of calling `s.NoError(e)`.
BTW, please use consistent name for errors. Per style used in the repo it should be `err` (not `e`), unless there's a reason.